Add plans doc

Itxaka opened the pull request:
Fixes #242 

Signed-off-by: Itxaka <igarcia@suse.com>

nunix commented on docs/elemental-plans.md:
```diff
@@ -0,0 +1,129 @@
+sets the proper hostname according to the `#!yaml MachineInventory`
+and installs the default Rancher System Agent from Rancher Server which then will install the proper Kubernetes components.
+
+This small bootstrap service also accepts local plans stored under `/var/lib/elemental/agent/plans` so any plan written
+in there will also be applied during the initial node start (After it has been installed).
```
```suggestion
The bootstrap service also accepts local plans stored under `/var/lib/elemental/agent/plans`. Any plan written
in there will also be applied during the initial node start after the installation is completed.
```

nunix commented on docs/elemental-plans.md:
```diff
@@ -0,0 +1,129 @@
+in there will also be applied during the initial node start (After it has been installed).
+
+!!! note
+    This local plans run only during the initial Elemental bootstrap **before** Kubernetes is installed on the node
```
```suggestion
    The local plans run only during the initial Elemental bootstrap **before** Kubernetes is installed on the node.
```

nunix commented on docs/elemental-plans.md:
```diff
@@ -0,0 +1,129 @@
+
+## Plan examples
+
+These plans are provided as a quick reference and not guaranteed to work. To learn more about plans please check [Rancher System Agent](https://github.com/rancher/system-agent)
```
```suggestion
The following plans are provided as a quick reference and are not guaranteed to work in your environment. To learn more about plans please check [Rancher System Agent](https://github.com/rancher/system-agent).
```

nunix commented on docs/elemental-plans.md:
```diff
@@ -0,0 +1,129 @@
+## Plan examples
+
+These plans are provided as a quick reference and not guaranteed to work. To learn more about plans please check [Rancher System Agent](https://github.com/rancher/system-agent)
+
```
```suggestion

**Example 1: one time instructions**

```

nunix commented on docs/elemental-plans.md:
````diff
@@ -0,0 +1,129 @@
+    ]
+}
+```
+
````
```suggestion

**Example 2: periodic instructions**
```

nunix commented on docs/elemental-plans.md:
````diff
@@ -0,0 +1,129 @@
+    ]
+}
+```
+
````
```suggestion

Example 3: files
```

nunix commented on docs/elemental-plans.md:
````diff
@@ -0,0 +1,129 @@
+    ]
+}
+```
+
````
```suggestion

Example 4: probes
```

nunix reviewed (commented):
Added introduction sentences for the examples and rephrased introduction.

Itxaka commented:
@nunix I added most of your comments except the examples header as I changed it to be an example block AND collapsed so they dont take the whole page and can be clicked to expand
